paging/src/commonMain/kotlin/dev/icerock/moko/paging/Pagination.kt

```diff
@@ -40,22 +40,32 @@ class Pagination<Item>(
     private val listMutex = Mutex()
 
     fun loadFirstPage() {
-        mEndOfList.value = false
-        mNextPageLoading.value = false
-
         launch {
-            listMutex.lock()
-            try {
-                val items: List<Item> = dataSource.loadPage(null)
-                mStateStorage.value = items.asState()
-            } catch (error: Throwable) {
-                mStateStorage.value = State.Error(error)
-            }
-            listMutex.unlock()
+            loadFirstPageSuspend()
+        }
+    }
+
+    suspend fun loadFirstPageSuspend() {
+        mEndOfList.postValue(false)
+        mNextPageLoading.postValue(false)
+
+        listMutex.lock()
```

Comment:
it should be before we change mEndOfList and mNextPageLoading

Reply:
Mutex lock was moved.